src/nnet3/nnet-utils.cc

```diff
@@ -610,8 +610,12 @@ class SvdApplier {
  public:
   SvdApplier(const std::string component_name_pattern,
              int32 bottleneck_dim,
+             BaseFloat energy_threshold,
```

Comment:
There needs to be some documentation here; it's not clear what these  parameters mean, how they interrelate, which takes priority over which, etc.  I also don't see parameter checking; what happens if people give invalid combinations of parameters?  Adding that would help clarify the code also.